biz.aQute.launcher/src/aQute/launcher/plugin/ProjectLauncherImpl.java

```diff
@@ -369,6 +369,10 @@ public Jar executable() throws Exception {
 		Manifest m = new Manifest();
 		Attributes main = m.getMainAttributes();
 
+		logger.debug("Use Embedded launcher");
+		main.putValue(MAIN_CLASS, EMBEDDED_LAUNCHER);
+		main.putValue(EMBEDDED_RUNPATH, join(classpath));
+
```

Comment:
Moving this up make it subject to damage by the REMOVEHEADERS work below. If for some reason REMOVEHEADERS contains MAIN_CLASS or EMBEDDED_RUNPATH, we are screwed now. I think we will need to make sure to remove MAIN_CLASS and EMBEDDED_RUNPATH from `result` before the call to `main.keySet().removeAll(result)` below.

Reply:
Should be fixed